.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,58 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Build Docker Image
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  pull_request:
+    branches: [ main ]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+#  workflow_dispatch:
+ #    branches:    
+ #     - main  
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build-image:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+    
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:        
+      # Start Docker Engine
+#          sudo systemctl start docker
+      - name: Start Docker Engine
+        run: |
+          sudo systemctl status docker -l
+          
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Listing Directory Contents
+        run: |
+          echo 'User:' $(whoami) 
+          echo 'Working Directory:' $(pwd) 
+          echo 'Directory Content:' 
+          ls -lah
+        
+      # Build Container Images
+      - name: Build Container Image with Docker
+        run: |
+          echo "Docker: Image building ..."
+          docker build -t forest -f ./Dockerfile .
```

Comment:
The Dockerfile probably also needs some work. I don't think anyone maintained that in a very long time.

Reply:
About the `Dockerfile` things that can be improved are a refined running environment for the daemon.
Now it runs as `root` which enables me to install any software on it:
```
$ docker exec -it 280c531fdb26 bash
root@280c531fdb26:/# netstat -lpn
bash: netstat: command not found
root@280c531fdb26:/# apt-get install net-tools
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  net-tools
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 248 kB of archives.
After this operation, 1002 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian buster/main amd64 net-tools amd64 1.60+git20180626.aebd88e-1 [248 kB]
Fetched 248 kB in 0s (944 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package net-tools.
(Reading database ... 18010 files and directories currently installed.)
Preparing to unpack .../net-tools_1.60+git20180626.aebd88e-1_amd64.deb ...
Unpacking net-tools (1.60+git20180626.aebd88e-1) ...
Setting up net-tools (1.60+git20180626.aebd88e-1) ...
root@280c531fdb26:/# netstat -lpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.1:6116          0.0.0.0:*               LISTEN      1/forest            
tcp        0      0 0.0.0.0:35761           0.0.0.0:*               LISTEN      1/forest            
tcp        0      0 127.0.0.1:1234          0.0.0.0:*               LISTEN      1/forest            
Active UNIX domain sockets (only servers)
Proto RefCnt Flags       Type       State         I-Node   PID/Program name     Path
```
So, I would suggest to create a non privilege user and a dedicated running environment.

Reply:
I opened a separate development at:
[Docker Deployment Improvement](https://github.com/ChainSafe/forest/issues/1445)
to further discuss possible improvements of the `Dockerfile` and their implications.